src/main/java/org/apache/commons/io/FileSystem.java

```diff
@@ -519,13 +528,11 @@ public boolean isLegalFileName(final CharSequence candidate) {
      * @since 2.21.0
      */
     public boolean isLegalFileName(final CharSequence candidate, final Charset charset) {
-        if (!isLegalFileLength(candidate, charset)) {
-            return false;
-        }
-        if (isReservedFileName(candidate)) {
-            return false;
-        }
-        return candidate.chars().noneMatch(this::isIllegalFileNameChar);
+        return candidate != null
+                && candidate.length() != 0
+                && nameLengthStrategy.isWithinLimit(candidate, getMaxFileNameLength(), charset)
```

Comment:
I think passing the candidate to the strategy would be better, this could then also implements things like “no more than 255 utf16 code units and no more than 300mbytes utf8 serialized”

Reply:
I am not sure either, probably missread the diff, please ignore the comment, looks good this way.